opensearch/src/main/java/org/opensearch/sql/opensearch/data/utils/OpenSearchJsonContent.java

```diff
@@ -126,28 +126,35 @@ public Object objectValue() {
   @Override
   public Pair<Double, Double> geoValue() {
     final JsonNode value = value();
-    if (value.has("lat") && value.has("lon")) {
-      Double lat = 0d;
-      Double lon = 0d;
+    Double lat = null;
+    Double lon = null;
+
+    if (value.has("lat")) {
       try {
         lat = extractDoubleValue(value.get("lat"));
       } catch (Exception exception) {
         throw new IllegalStateException(
             "latitude must be number value, but got value: " + value.get(
                 "lat"));
       }
+    }
+
+    if (value.has("lon")) {
       try {
         lon = extractDoubleValue(value.get("lon"));
       } catch (Exception exception) {
         throw new IllegalStateException(
             "longitude must be number value, but got value: " + value.get(
                 "lon"));
       }
-      return Pair.of(lat, lon);
-    } else {
-      throw new IllegalStateException("geo point must in format of {\"lat\": number, \"lon\": "
+    }
+
+    if (lat == null && lon == null) {
```

Comment:
Are you allowed to define a geo point with only a longitude or latitude? Because according to this code it appears that it only throws an exception if both are `null`